src/main/scripts/src/app/shared/perspectives/documents/post-it/post-it-documents-perspective.component.ts

```diff
@@ -18,23 +18,150 @@
  * -----------------------------------------------------------------------/
  */
 
-import {Component, Input} from '@angular/core';
+import {animate, style, transition, trigger} from '@angular/animations';
+import {ActivatedRoute, ParamMap} from '@angular/router';
+import {Component, Input, OnInit} from '@angular/core';
+
 import {DocumentService} from '../../../../core/rest/document.service';
+import {WorkspaceService} from '../../../../core/workspace.service';
+import {LocalService} from '../../../../core/local.service';
 import {Perspective} from '../../perspective';
+import {CollectionService} from '../../../../core/rest/collection.service';
+import {Collection} from '../../../../core/dto/collection';
+import {Document} from '../../../../core/dto/document';
 
 @Component({
   selector: 'post-it-documents-perspective',
-  templateUrl: './post-it-documents-perspective.component.html'
+  templateUrl: './post-it-documents-perspective.component.html',
+  styleUrls: ['./post-it-documents-perspective.component.scss'],
+  animations: [
+    trigger('appear', [
+      transition(':enter', [
+        style({transform: 'scale(0)'}),
+        animate('0.25s ease-out', style({transform: 'scale(1)'})),
+      ]),
+      transition(':leave', [
+        style({transform: 'scale(1)'}),
+        animate('0.25s ease-out', style({transform: 'scale(0)'})),
+      ])
+    ])
+  ]
 })
-export class PostItDocumentsPerspectiveComponent implements Perspective {
+export class PostItDocumentsPerspectiveComponent implements Perspective, OnInit {
 
   @Input()
   public query: string;
 
   @Input()
   public editable: boolean;
 
-  constructor(private documentService: DocumentService) {
+  @Input()
+  public height: string = '500px';
+
+  public collection: Collection;
+
+  public addDocumentAttribute: string;
+  public addButtonColor: string;
+  public cursorOnAddButton: boolean;
+
+  public colors: string[];
+  public icons: string[];
+  public documents: Document[];
+
+  constructor(private documentService: DocumentService,
+              private collectionService: CollectionService,
+              private workspaceService: WorkspaceService,
+              private route: ActivatedRoute,
+              private localService: LocalService) {
+  }
+
+  public ngOnInit() {
+    // initialize workspace if needed
+    let setWorkspaceParams = (params: ParamMap) => {
+      this.workspaceService.projectCode = params.get('projectCode');
+      this.workspaceService.organizationCode = params.get('organizationCode');
+    };
+
+    if (!this.workspaceService.isWorkspaceSet()) {
+      this.route.paramMap.subscribe(setWorkspaceParams);
+    }
+
+    // get collection
```

Comment:
You can extract some parts of `ngOnInit` method to multiple methods in order to increase code readability and avoid comments.